modules/claims_api/app/services/claims_api/dependent_claimant_poa_assignment_service.rb

```diff
@@ -0,0 +1,169 @@
+
+      log(level: :error, detail: 'Failed to assign POA to dependent')
+
+      raise ::Common::Exceptions::FailedDependency
+    end
+
+    private
+
+    def person_web_service
+      ClaimsApi::PersonWebService.new(external_uid: @dependent_participant_id,
+                                      external_key: @dependent_participant_id)
+    end
+
+    def log(level: :info, **rest)
+      ClaimsApi::Logger.log('dependent_claimant_poa_assignment_service',
+                            level:, dependent_participant_id: @dependent_participant_id,
+                            poa_code: @poa_code, veteran_participant_id: @veteran_participant_id, **rest)
+    end
+
+    def assign_poa_to_dependent_via_manage_ptcpnt_rlnshp?
+      res = person_web_service.manage_ptcpnt_rlnshp_poa(ptcpnt_id_a: @dependent_participant_id,
+                                                        ptcpnt_id_b: poa_participant_id)
+
+      if manage_ptcpnt_rlnshp_poa_success?(res)
+        log(detail: 'POA assigned to dependent.')
+
+        return true
+      end
+
+      log(level: :warn,
+          detail: 'Something else went wrong with manage_ptcpnt_rlnshp. Falling back to update_benefit_claim.')
+
+      false
+    rescue ::Common::Exceptions::ServiceError => e
+      if e.errors.first.detail == 'PtcpntIdA has open claims.'
+        log(detail: 'Dependent has open claims, continuing.')
+      else
+        log(level: :warn,
+            detail: 'Something else went wrong with manage_ptcpnt_rlnshp. Falling back to update_benefit_claim.')
+      end
+
+      false
+    end
+
+    def iso_to_date(iso_date)
+      DateTime.parse(iso_date).strftime('%m/%d/%Y')
+    end
+
+    def build_benefit_claim_update_input(claim_details:)
+      claim_rcvd_dt = iso_to_date(claim_details[:claim_rcvd_dt])
+
+      {
+        file_number: @veteran_file_number,
+        payee_code: claim_details[:payee_type_cd],
+        date_of_claim: claim_rcvd_dt,
+        claimant_ssn: @claimant_ssn,
+        power_of_attorney: @poa_code,
+        benefit_claim_type: benefit_claim_type(claim_details[:pgm_type_cd]),
+        old_end_product_code: claim_details[:cp_claim_end_prdct_type_cd],
+        new_end_product_label: claim_details[:bnft_claim_type_cd],
+        old_date_of_claim: claim_rcvd_dt,
+        allow_poa_access: @allow_poa_access,
+        allow_poa_cadd: @allow_poa_cadd
+      }
+    end
+
+    def assign_poa_to_dependent_via_update_benefit_claim?
+      first_open_claim = dependent_claims.find do |claim|
+        claim[:phase_type] != 'Complete' && claim[:ptcpnt_vet_id] == @veteran_participant_id
+      end
+      first_open_claim_details = claim_details(first_open_claim[:benefit_claim_id])
+
+      benefit_claim_update_input = build_benefit_claim_update_input(claim_details: first_open_claim_details)
+
+      result = benefit_claim_service.update_benefit_claim(benefit_claim_update_input)
+
+      if result[:return][:return_message] == 'Update to Corporate was successful'
```

Comment:
Brittle check to verify BGS update succeeded (a slight change in the message would break the service). Any better suggestions?

Reply:
All things being equal, I prefer keying off the return message since it makes the code more readable. 🤷

Reply:
This also makes me a bit nervous since we don't control the message, `Update to Corporate was successful`, and it could change. It also assumes the structure of `result` always has `return` and `return_message`. At the very least, would this be a slight improvement? 
```ruby
if result.dig(:return, :return_message) == 'Update to Corporate was successful'
```

Reply:
Sorry - just now seeing these. If the response structure changes, I'm not sure digging will help. If anything, we would want to fail loudly here to investigate (likely our only way of telling BGS changed).